protocol/prottest/block.go

```diff
@@ -15,27 +16,46 @@ import (
 )
 
 var (
-	mutex  sync.Mutex // protects the following
-	states = make(map[*protocol.Chain]*state.Snapshot)
+	mutex         sync.Mutex // protects the following
+	states        = make(map[*protocol.Chain]*state.Snapshot)
+	blockPubkeys  = make(map[*protocol.Chain][]ed25519.PublicKey)
+	blockPrivkeys = make(map[*protocol.Chain][]ed25519.PrivateKey)
 )
 
-type Option func(*config)
+type Option func(testing.TB, *config)
 
 func WithStore(store protocol.Store) Option {
-	return func(conf *config) { conf.store = store }
+	return func(_ testing.TB, conf *config) { conf.store = store }
 }
 
 func WithOutputIDs(outputIDs ...bc.Hash) Option {
-	return func(conf *config) {
+	return func(_ testing.TB, conf *config) {
 		for _, oid := range outputIDs {
 			conf.initialState.Tree.Insert(oid.Bytes())
 		}
 	}
 }
 
+func WithBlockSigners(quorum, n int) Option {
+	return func(tb testing.TB, conf *config) {
+		conf.quorum = quorum
+		for i := 0; i < n; i++ {
+			pubkey, privkey, err := ed25519.GenerateKey(nil)
```

Comment:
It is a minor bummer that GenerateKey doesn't just panic if it can't get randomness.